### ISSUE TITLE
Better support MultiPoint, MultiLineString, and MultiPolygon

### ISSUE THREAD

**Issue by supersonicclay** [uber/nebula.gl#42]

**Comment by supersonicclay** [comment]
Notes on the subject:

I think we'll need a term like "tentative feature". That makes it clear that it isn't a real thing yet. And I think it'd only apply to draw* modes. It'll need to stick around between `onEdit` calls, so it can't just be the `updatedData` argument. It'll need its own state/prop.

Here's my thought:

```
new EditableGeoJsonLayer({
  data: this.state.data,

  // optional if caller wants to customize tentative feature handling
  tentativeFeature: this.state.tentativeFeature, 

  onEdit: ...,

  // optional if caller wants to customize tentative feature handling
  onTentativeFeatureUpdate: ({updatedTentativeFeature}) => {
      tentativeFeature: updatedTentativeFeature,
  },
})
```


A couple of examples.

## Draw new line string

* selectedFeatureIndexes=[]
* mode=drawLineString
* user moves pointer around
  * onTentativeFeatureUpdate called repeatedly as pointer moves around
    * updatedTentativeFeature is a point feature whose coordinates follow the mouse
* user clicks
  * onEdit is not called because it isn't a LineString yet
  * onTentativeFeatureUpdate called repeatedly as pointer moves around
    * updatedTentativeFeature is a LineString with first coordinate where user clicked, second coordinate follows the mouse
* user clicks
  * onEdit called, updatedData has a new LineString feature, updatedSelectedIndexes would have the new one selected
  * onTentativeFeatureUpdate called repeatedly as pointer moves around
    * updatedTentativeFeature is a LineString with first coordinate where user clicked the second time, second coordinate follows the mouse


## Draw new line polygon

* selectedFeatureIndexes=[]
* mode=drawPolygon
* user moves pointer around
  * onTentativeFeatureUpdate called repeatedly as pointer moves around
    * updatedTentativeFeature is a point feature whose coordinates follow the mouse
* user clicks
  * onEdit is not called because it isn't a Polygon yet
  * onTentativeFeatureUpdate called repeatedly as pointer moves around
    * updatedTentativeFeature is a LineString with first coordinate where user clicked, second coordinate follows the mouse
* user clicks
  * onEdit is not called because it isn't a Polygon yet
  * onTentativeFeatureUpdate called repeatedly as pointer moves around
    * updatedTentativeFeature is a Polygon (triangle) with first coordinate where user clicked the first time, the second coordinate where the user clicked the second time, the third coordinate follows the mouse, and the fourth coordinate loops back to the first coordinate (completing the triangle)
* user clicks
  * onEdit called, updatedData has a new Polygon feature, updatedSelectedIndexes would have the new one selected
    * updatedTentativeFeature is a new a Point again following the mouse in order to add another Polygon to the feature (thus upgrading it to a MultiPolygon)

**Comment by supersonicclay** [comment]
MultiPolygon support implemented in #102

**Comment by titivermeesch** [comment]
Maybe this is already added, but is there a MultiLineString mode?

**Comment by georgios-uber** [comment]
@titivermeesch it is supported to be edited but not created.

**Comment by titivermeesch** [comment]
@georgios-uber So now way of creating a path at the moment? Or am I using the feature wrong. The main purpose is to make a path between 2 locations following a road for example

**Comment by supersonicclay** [comment]
You can use `DrawLineStringMode`. If you set `selectedFeatureIndexes=[]` then it'll draw the first segment. But after you get the first call to `onEdit`, then you want to set the `selectedFeatureIndexes` prop to the new feature index in order to extend that line.

**Comment by titivermeesch** [comment]
I see how that would work. But does that mean that the user will have to start drawing every time or will the drawing mode remain so the user can continue drawing a line without stopping

**Comment by supersonicclay** [comment]
Yes, users can draw a continuous line string. Probably the best way forward is to see it in action at https://nebula.gl/geojson-editor/ and select "Draw LineString" on the left side.

![draw-linestring](https://user-images.githubusercontent.com/469582/73149963-12e24900-4081-11ea-979f-b43e273d5e67.gif)
